pyquil/api/_wavefunction_simulator.py

```diff
@@ -230,4 +230,4 @@ def augment_program_with_memory_values(
 
         p += quil_program
 
-        return percolate_declares(p)
+        return p
```

Comment:
As of v4, `percolate_declares` is a no-op (quil-rs does that automatically), so I removed its usage here.